src/overlay/Peer.h

```diff
@@ -212,7 +213,7 @@ class Peer : public std::enable_shared_from_this<Peer>,
 
     std::string mRemoteVersion;
     uint32_t mRemoteOverlayMinVersion;
-    uint32_t mRemoteOverlayVersion;
+    std::optional<uint32_t> mRemoteOverlayVersion;
```

Comment:
Made this optional instead of 0, so that we don't drop pending peers when we don't know their version yet (this can happen if we've sent HELLO, but haven't received a response)